purchases/index.js

```diff
@@ -528,28 +529,50 @@ function listPurchases(db, options, callback) {
     });
 }
 
+function createWhereStatement(options) {
+    const parts = [parseCategory(options), parseAfter(options), parseBefore(options)].filter(x => x !== "");
+    if(parts.length === 0) {
+        return "";
+    }
+    return `WHERE ${parts.join(" AND ")}`;
+}
+
 function parseCategory({categories}) {
     if(!categories) {
         return "";
     }
     categoriesAsArray = categories.split(",");
-    return `WHERE status IN (${categoriesAsArray.map(category => `'${category}'`).join(", ")})`;   
+    return `status IN (${categoriesAsArray.map(category => `'${category}'`).join(", ")})`;   
+}
+
+function parseAfter({after}) {
+    if(!after) {
+        return "";
+    }
+    return `(data ->> 'completedAt')::timestamptz > '${after}'::timestamptz`;
+}
+
+function parseBefore({before}) {
+    if(!before) {
+        return "";
+    }
+    return `(data ->> 'completedAt')::timestamptz <= '${before}'::timestamptz`;
 }
 
 function parseLimit({ limit }) {
     const limitAsInt = parseInt(limit);
-    if(limitAsInt > 0 && !isNaN(limitAsInt)) {
-        return `LIMIT ${limitAsInt}`;
+    if(isNaN(limitAsInt) || limitAsInt < 0) {
```

Comment:
Prefer `Number.isNaN` for clarity. Avoid global scope built-in functions when possible.